google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -993,6 +993,56 @@ public void testListBlobsVersioned() throws ExecutionException, InterruptedExcep
     }
   }
 
+  @Test
+  public void testListBlobsWithOffset() throws ExecutionException, InterruptedException {
+    String bucketName = RemoteStorageHelper.generateBucketName();
+    Bucket bucket =
+        storage.create(BucketInfo.newBuilder(bucketName).setVersioningEnabled(true).build());
+    try {
+      String[] blobNames = {
+        "test-list-blobs-start-offset-blob1",
+        "test-list-blobs-start-offset-blob2",
+        "test-list-blobs-end-offset-blob3"
+      };
+      BlobInfo blob1 =
+          BlobInfo.newBuilder(bucket, blobNames[0]).setContentType(CONTENT_TYPE).build();
+      BlobInfo blob2 =
+          BlobInfo.newBuilder(bucket, blobNames[1]).setContentType(CONTENT_TYPE).build();
+      BlobInfo blob3 =
+          BlobInfo.newBuilder(bucket, blobNames[2]).setContentType(CONTENT_TYPE).build();
+
+      Blob remoteBlob1 = storage.create(blob1);
+      Blob remoteBlob2 = storage.create(blob2);
+      Blob remoteBlob3 = storage.create(blob3);
+      assertNotNull(remoteBlob1);
+      assertNotNull(remoteBlob2);
+      assertNotNull(remoteBlob3);
+      Page<Blob> page =
+          storage.list(
+              bucketName,
+              Storage.BlobListOption.startOffset("test-list-blobs-start-offset-blob"),
+              Storage.BlobListOption.endOffset("test-list-blobs-end-offset-blob3"));
```

Comment:
Recommend starting from not the first object in the new objects.

Reply:
Replace this test with new one.